@Hang-Lei-NOAA I think the C test error is not due to the change in this PR.
The error information is as follows:
CMake Error at /usr/local/share/cmake-3.27/Modules/CMakeDetermineFortranCompiler.cmake:33 (message):
CMake Error: CMAKE_Fortran_COMPILER not set, after EnableLanguage

This failed test is for Intel. Do we have an Intel compiler available in Github?

Upon further checking, it looks like the process of downloading and installing an Intel compiler failed. If someone can rerun the tests, it may pass?

--- a/.github/workflows/Intel.yml
+++ b/.github/workflows/Intel.yml
@@ -35,7 +35,7 @@ jobs:
         rm GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
         echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
         sudo apt-get update
-        sudo apt-get install intel-oneapi-dev-utilities intel-oneapi-mpi-devel intel-oneapi-openmp intel-oneapi-compiler-fortran intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic
+        sudo apt-get install intel-oneapi-dev-utilities intel-oneapi-mpi-devel intel-oneapi-openmp intel-oneapi-compiler-fortran-2023.2.1 intel-oneapi-compiler-dpcpp-cpp-and-cpp-classi
         echo "source /opt/intel/oneapi/setvars.sh" >> ~/.bash_profile
 
     - name: cache-bacio

--- a/ush/err_exit
+++ b/ush/err_exit
@@ -67,6 +67,3 @@ if [ ! -z $PBS_JOBID ]; then
 elif [ ! -z $SLURM_JOB_ID ]; then
   scancel $SLURM_JOB_ID
 fi
-else
-  ecflow_client --kill=${ECF_NAME:?}
-fi

--- a/ush/err_exit
+++ b/ush/err_exit
@@ -58,10 +58,15 @@ if [ "$SENDECF" = "YES" ]; then
 fi
 
 # KILL THE JOB:
+if [ "$SENDECF" = "YES" ]; then
+  ecflow_client --kill=${ECF_NAME:?}
+fi
+
 if [ ! -z $PBS_JOBID ]; then
   qdel $PBS_JOBID
 elif [ ! -z $SLURM_JOB_ID ]; then
   scancel $SLURM_JOB_ID
+fi
 else
   ecflow_client --kill=${ECF_NAME:?}
 fi

